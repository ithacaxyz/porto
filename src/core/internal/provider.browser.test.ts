import { describe, expect, test } from 'vitest'

import { porto, run } from '../../../test/utils.browser.js'

describe('wallet_connect', () => {
  test('sign up', async () => {
    const { accounts } = await run(
      porto.provider.request({ method: 'wallet_connect' }),
      (iframe) => iframe.getByTestId('sign-up').click(),
    )
    expect(accounts.length).toBe(1)
  })
})
