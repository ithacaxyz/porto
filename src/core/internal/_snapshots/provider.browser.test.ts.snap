// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`wallet_getPermissions > behavior: grant on connect; grant another; get after connect 1`] = `
[
  {
    "address": null,
    "chainId": null,
    "expiry": 9999999999,
    "id": "0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
    "key": {
      "publicKey": "0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
      "type": "secp256k1",
    },
    "permissions": {
      "calls": [
        {
          "signature": "mint()",
        },
      ],
    },
  },
  {
    "address": null,
    "chainId": null,
    "expiry": 9999999999,
    "id": "0xcafebabecafebabecafebabecafebabecafebabecafebabecafebabecafebabe",
    "key": {
      "publicKey": "0xcafebabecafebabecafebabecafebabecafebabecafebabecafebabecafebabe",
      "type": "p256",
    },
    "permissions": {
      "calls": [
        {
          "signature": "mint()",
        },
      ],
    },
  },
]
`;

exports[`wallet_getPermissions > behavior: grant on connect; grant another; get after connect 2`] = `
[
  {
    "address": null,
    "chainId": null,
    "expiry": 9999999999,
    "id": "0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
    "key": {
      "publicKey": "0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
      "type": "secp256k1",
    },
    "permissions": {
      "calls": [
        {
          "signature": "mint()",
        },
      ],
    },
  },
  {
    "address": null,
    "chainId": null,
    "expiry": 9999999999,
    "id": "0xcafebabecafebabecafebabecafebabecafebabecafebabecafebabecafebabe",
    "key": {
      "publicKey": "0xcafebabecafebabecafebabecafebabecafebabecafebabecafebabecafebabe",
      "type": "p256",
    },
    "permissions": {
      "calls": [
        {
          "signature": "mint()",
        },
      ],
    },
  },
]
`;
